### PR TITLE
Interactive Video V1 JIRA key-> SB-26330

### PR DESCRIPTION
**Pull request by Harithaa95**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-26330" title="SB-26330" target="_blank">StorySB-26330</a>  Creator Interface for Interactive Videos
  </td></table>
  <br />
 

[SB-26330](https://project-sunbird.atlassian.net/browse/SB-26330) Interactive Video V1

### CODE REVIEW

File: src/app/client/src/app/modules/sourcing/services/quml-player/quml-player.service.ts [hunk 1]
```diff
@@ -43,11 +47,24 @@ export class QumlPlayerService implements QuestionCursor, EditorCursor {
...
+     // if (this.listUrl) {
+           const hierarchy =  this.http.get("action/questionset/v1/hierarchy/"+identifier+"?mode=edit")
+           const questionSet = this.http.get(`action/questionset/v1/read/${identifier}?fields=instructions`)
+            return (
+                forkJoin([hierarchy, questionSet]).pipe(
+                    map(res => {
+                        let questionSet =  res[0]['result']['questionSet'];
+                        if(res[1]['result']['questionset']['instructions'] && questionSet) {
+                            questionSet['instructions']
+                        }
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/services/quml-player/quml-player.service.ts:59]: please add error check to get nested properties. for example: use `_.get(res[0], 'result.questionSet')` and `_.get(res[1], 'result.questionset.instructions')`

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.spec.ts [hunk 2]
```diff
@@ -452,4 +454,72 @@ xdescribe('ContentUploaderComponent', () => {
       expect(helperServiceStub.submitRequestChanges).not.toHaveBeenCalled();
   });
 
+  /***********************************************/
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.spec.ts:457]: remove these lines

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.html [hunk 3]
```diff
@@ -105,16 +111,56 @@
...
+         <div class="sb-modal-content">
+            <form #FormControl="ngForm" class="ui form">
+               <div class = "lableDiv">Question TimeStamp</div>
+                  <div class="d-flex">
+                     <div class="timeDiv">
+                           <input class="sb-form-control-tab" placeholder="00:00" [(ngModel)]="interceptionTime" name="interceptionTime">
+                        
+                       <!--  <div *ngFor = "let time of interceptionPoint" >
+                              <input maxlength=1 class="sb-form-control-tab" [(ngModel)]="time" />
+                        </div> -->
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.html:149]: remove the commented lines

File: src/app/client/src/app/app.module.ts [hunk 4]
```diff
@@ -41,6 +42,7 @@ import { QumlPlayerService } from './modules/sourcing/services/quml-player/quml-
     ...PluginModules,
     QumlLibraryModule,
     CollectionEditorLibraryModule,
+    SunbirdVideoPlayerModule,
```
Comment (venkateshwarans) [inline src/app/client/src/app/app.module.ts:45]: we don't have to import the video player module at the app level. only in sourcing module is enough

File: src/app/client/src/app/modules/shared/services/config/app.config.json [hunk 5]
```diff
@@ -725,6 +725,10 @@
           "id": "sunbird.portal",
           "ver": "1.8.0",
           "pid": "sunbird-portal"
+        },
+        "userData": {
+          "firstName": "Sourav",
+          "lastName": "Dey"
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/shared/services/config/app.config.json:731]: can you make the userData as empty object?

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.html [hunk 6]
```diff
@@ -105,16 +111,56 @@
         </div>
   </div>
 </div>
-<div class="middle-section-set" *ngIf="showPreview">
-    <div class="ui container msg-box">
-      <div class="msg-box__content">
-        <div class="container msg-box__content__blank-box">
-            <app-player class="content-player" [playerConfig]="playerConfig"></app-player>
+<!-- /****************************************************/ -->
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.html:114]: please move these asterisk lines everywhere

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.html [hunk 7]
```diff
@@ -105,16 +111,56 @@
...
+        <button  class="sb-btn sb-btn-outline-primary sb-btn-normal mr-8" id="addQuestionSet"
+        suiPopup popupText="Add a new Question" [popupPlacement]="'top'" (click)="addInterception()">
+                Add a new Question
+        </button>
+      </div>
     </div>
+  </div>
 </div>
-
+<sui-modal *ngIf="showquestionCreationUploadModal" [mustScroll]="true" [isClosable]="true" [transitionDuration]="0" [size]="'normal'"
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.html:137]: please maintain consistency in naming variables. use proper camelCase for `showquestionCreationUploadModal`. make it `showQuestionCreationUploadModal`

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts [hunk 8]
```diff
@@ -19,7 +19,20 @@ import { IStartEventInput, IEndEventInput, TelemetryService } from '@sunbird/tel
...
+import { ProgramComponentsService } from '../../../program/services/program-components/program-components.service';
+import { InitialState } from '../../interfaces';
+/******************************/
+/******************************/
+interface IDynamicInput {
+  contentUploadComponentInput?: IContentUploadComponentInput;
+  resourceTemplateComponentInput?: IResourceTemplateComponentInput;
+  practiceQuestionSetComponentInput?: any;
+  contentEditorComponentInput?: IContentEditorComponentInput;
+  questionSetEditorComponentInput?: any;
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts:33]: we only need `questionSetEditorComponentInput` inside `IDynamicInput`. please remove the rest of the properties.

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts [hunk 9]
```diff
@@ -113,6 +126,118 @@ export class ContentUploaderComponent implements OnInit, AfterViewInit, OnDestro
...
+  //   data: {}
+  // };
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts:240]: Please remove the commented out code

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts [hunk 10]
```diff
@@ -1136,4 +1297,1092 @@ export class ContentUploaderComponent implements OnInit, AfterViewInit, OnDestro
...
+          'X-Channel-Id': this.programContext.rootorg_id
+        },
+        data: {
+          request: {
+            questionset: {
+              'name': 'Untitled',
+              'code': UUID.UUID(),
+              'mimeType': "application/vnd.sunbird.questionset",
+              'createdBy': this.userService.userid,
+              'primaryCategory': "Demo Practice Question Set",
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts:1419]: @Harithaa95  are we going to hardcode the primaryCategory as `Demo Practice Question Set`? 
CC @pallakartheekreddy

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts [hunk 11]
```diff
@@ -750,9 +906,9 @@ export class ContentUploaderComponent implements OnInit, AfterViewInit, OnDestro
         this.resourceStatusClass = 'sb-color-gray-300';
       }
 
-      this.playerConfig = this.playerService.getConfig(contentDetails);
-      this.playerConfig.context.pdata.pid = `${this.configService.appConfig.TELEMETRY.PID}`;
-      this.playerConfig.context.cdata = _.get(this.sessionContext, 'telemetryPageDetails.telemetryInteractCdata') || [];
+      this.playerConfigSunbird = this.playerService.getConfig(contentDetails);
+      this.playerConfigSunbird.context.pdata.pid = `${this.configService.appConfig.TELEMETRY.PID}`;
+      this.playerConfigSunbird.context.cdata = _.get(this.sessionContext, 'telemetryPageDetails.telemetryInteractCdata') || [];
```
Comment (venkateshwarans) [inline src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts:911]: Any particular reason to rename the `this.playerConfig` variable? Can you revert back to `this.playerConfig`?

File: src/app/client/package.json [hunk 12]
```diff
@@ -42,12 +42,13 @@
...
     "@project-sunbird/common-consumption-v8": "3.6.4",
-    "@project-sunbird/sunbird-collection-editor-v9": "4.3.18",
-    "@project-sunbird/sunbird-pdf-player-v9": "4.1.1",
-    "@project-sunbird/sunbird-quml-player-v9": "4.3.9",
-    "@project-sunbird/sunbird-video-player-v9": "4.1.1",
+    "@project-sunbird/sb-styles": "0.0.9",
+    "@project-sunbird/sunbird-collection-editor-v9": "4.3.7",
+    "@project-sunbird/sunbird-pdf-player-v9": "4.2.0",
+    "@project-sunbird/sunbird-quml-player-v9": "4.3.3",
+    "@project-sunbird/sunbird-video-player-v9": "4.3.0",
```
Comment (venkateshwarans) [inline src/app/client/package.json:51]: Please change all the library package versions to recent versions. For example - you are downgrading  `@project-sunbird/sunbird-collection-editor-v9` from  `4.3.18` to `4.3.7`. keep it as `4.3.18`. and do the same of the rest

**Review by venkateshwarans — CHANGES_REQUESTED** [review]
Lots of changes required. And quite of lot of unnecessary code has been introduced to `content-uploader.compontent.ts`. please move them and make sure we test locally before pushing the changes.

File: src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts (hunk 11 above)
Comment (Harithaa95) [reply src/app/client/src/app/modules/sourcing/components/content-uploader/content-uploader.component.ts:911]: Already playerConfig Variable is there and used by contentuploader component. So introduced new variable for question set editor.

**Pull request closed without merging**